service/history/workflow/cache/cache_test.go

```diff
@@ -530,3 +531,216 @@ func (s *workflowCacheSuite) TestCacheImpl_lockWorkflowExecution() {
 		})
 	}
 }
+
+func (s *workflowCacheSuite) TestCacheImpl_SizeBasedCacheBasic() {
```

Comment:
Can you break this test apart into multiple tests that check specific behaviors rather than one test for all behaviors? For example:

- `TestWorkflowCache_EnforcesMaxSizeLimitOnRelease`
- `TestWorkflowCache_RejectsRequestWhenAtLimit`

etc.